lib/dispatcher.rb

```diff
@@ -28,7 +28,17 @@ def on_approval_status_change(approval)
   end
 
   def requires_approval_notice?(approval)
-    true
+    if approval.cart.getProp('origin') == 'ncr'
+      self.with_approval_notice_requirements { approval.is_final_approval? }
```

Comment:
+1 separating this out. Ideally, the code should never explicitly check "am I ncr?". Calling could be based on method overriding or configurations. For the latter, I'd imagine an api like this:

```ruby
if approval.cart.getProp('origin').needs_notice
```